### PR TITLE
feat: Player Armor Hiding

### PR DESCRIPTION
**Pull request by P0keDev**
Allows the user to disable armor rendering on a piece-by-piece basis (helmets, chestplates, etc.) and offers a separate option to preserve Wynn cosmetic rendering. This feature does not conflict with other armor-related features like the gear viewer or the nametag gear list. I decided to leave this feature disabled by default as it's a relatively drastic change to the vanilla Wynn experience and I'm unsure how popular it might be with the average player.

To implement this feature I needed an armor rendering event that fired once per armor piece instead of once per entity, which is how `PlayerArmorRenderEvent` functioned previously; I decided to repurpose this event instead of creating a separate event, but I can revert that change if necessary. Since certain head rendering is done separately by `CustomHeadLayer`, I had to mix into that rendering class as well, but I decided to use the same event for it as it fundamentally accomplishes the same thing.

While working on these render events, I noticed a bug in `PlayerGhostTransparencyFeature` which caused the value of the transparent armor setting to be ignored, so I fixed that as well.

In this example, boots are enabled and helmets are not, but the cosmetic override option is enabled. Also showcases that the nametag feature is unaffected.
image

### CODE REVIEW

**Comment by kristofbolyai** [comment]
Can't we remove the setting from player ghost transparency feature? This is essentially the same thing isn't it? 

Also, how do you feel about merging this feature into pghost transparency? (just an idea, wanna hear your opinion)

**Comment by kristofbolyai** [comment]
Sounds solid though

**Comment by P0keDev** [comment]
removing the ghost armor setting would still limit player options if, say, they wanted to see certain pieces of armor on non-ghosts but still remove armor from ghosts. maybe it would be cleaner to move the ghost armor setting into this feature though?

**Comment by kristofbolyai** [comment]
> removing the ghost armor setting would still limit player options if, say, they wanted to see certain pieces of armor on non-ghosts but still remove armor from ghosts. maybe it would be cleaner to move the ghost armor setting into this feature though?

Oh right because this works for non ghosts. Then this should be fine as is. I'll review throughly tomorrow, I am off to bed today

File: common/src/main/java/com/wynntils/features/user/players/PlayerArmorHidingFeature.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+    @Config
+    public boolean showCosmetics = true;
+
+    @SubscribeEvent
+    public void onPlayerArmorRender(PlayerArmorRenderEvent event) {
+        switch (event.getSlot()) {
+            case HEAD -> {
+                if (!hideHelmets) return;
+                if (!showCosmetics) { // helmet is hidden regardless, no extra logic needed
```
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/features/user/players/PlayerArmorHidingFeature.java:40]: Can you return early here too? Just a nit though

**Review by kristofbolyai — APPROVED** [review]
Looks good :)

File: common/src/main/java/com/wynntils/features/user/players/PlayerArmorHidingFeature.java (hunk 1 above)
Comment (P0keDev) [reply common/src/main/java/com/wynntils/features/user/players/PlayerArmorHidingFeature.java:40]: where, exactly? this `if` statement does return, it just needs to cancel the event first

Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/features/user/players/PlayerArmorHidingFeature.java:40]: if showcosmetics return;
e.cancel

Comment (P0keDev) [reply common/src/main/java/com/wynntils/features/user/players/PlayerArmorHidingFeature.java:40]: not every helmet is a cosmetic though - that logic wouldn't match the functionality

Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/features/user/players/PlayerArmorHidingFeature.java:40]: oh, leave it then

**Pull request merged**